stream.go

```diff
@@ -155,6 +155,15 @@ func (cs *clientStream) Trailer() metadata.MD {
 }
 
 func (cs *clientStream) SendMsg(m interface{}) (err error) {
+	if EnableTracing {
+		cs.traceInfo.tr.LazyLog(payload{m}, true)
```

Comment:
On Thu, Jun 25, 2015 at 5:23 PM, David Symonds <notifications@github.com>
wrote:

> In stream.go
> <https://github.com/grpc/grpc-go/pull/235#discussion_r33319598>:
>
> > @@ -155,6 +155,15 @@ func (cs *clientStream) Trailer() metadata.MD {
> >  }
> >
> >  func (cs *clientStream) SendMsg(m interface{}) (err error) {
> > +	if EnableTracing {
> > +		cs.traceInfo.tr.LazyLog(payload{m}, true)
>
> I don't think the particular use of payloads for streaming should matter.
> Unary RPCs can be just as big. Controls over the size are a fine thing to
> have, but that's independent of this change.
>
> (There's no need for control on the number of payloads; the trace package
> will only hold on to 10 events per trace. The only constraint may be on a
> per-payload basis, which will be a little harder to do since we permit
> arbitrary codecs. I have ideas for how to handle that though.)
>
So a trace hold up to 10 payload for an RPC? The latest will overwrite the
oldest? If that is the case, it sgtm. Otherwise it could be much worse than
unary rpc (infinite * 1 payload vs. 1 payload).

BTW, does trace truncate the payload if it is large?

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/235/files#r33319598>.
>